src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -532,94 +513,100 @@ public String getLoginUrl() {
-                    }
-                });
-        GenericUrl url = new GenericUrl(getDefaultedServerPrefix()
-                + OAUTH_PROVIDER_URI);
+        HttpRequestFactory requestFactory = transport.createRequestFactory(new HttpRequestInitializer() {
+            public void initialize(HttpRequest request) throws IOException {
+                credential.initialize(request);
+                request.setParser(new JsonObjectParser(JSON_FACTORY));
+            }
+        });
+        GenericUrl url = new GenericUrl(getDefaultedServerPrefix() + OAUTH_PROVIDER_URI);
 
         HttpRequest request = requestFactory.buildGetRequest(url);
 
-        OpenShiftProviderInfo info = request.execute().parseAs(
-                OpenShiftProviderInfo.class);
+        OpenShiftProviderInfo info = request.execute().parseAs(OpenShiftProviderInfo.class);
         return info;
     }
-    
+
     private HttpTransport transportToUse(final Credential credential) {
         if (provider == null)
             return transport;
         try {
-            HttpRequestFactory requestFactory = transport
-                    .createRequestFactory(new HttpRequestInitializer() {
-                        public void initialize(HttpRequest request)
-                                throws IOException {
-                            credential.initialize(request);
-                            request.setParser(new JsonObjectParser(JSON_FACTORY));
-                        }
-                    });
+            HttpRequestFactory requestFactory = transport.createRequestFactory(new HttpRequestInitializer() {
+                public void initialize(HttpRequest request) throws IOException {
+                    credential.initialize(request);
+                    request.setParser(new JsonObjectParser(JSON_FACTORY));
+                }
+            });
             GenericUrl url = new GenericUrl(provider.token_endpoint);
             HttpRequest request = requestFactory.buildHeadRequest(url);
             request.execute().getStatusCode();
         } catch (com.google.api.client.http.HttpResponseException hre) {
-            if (hre.getStatusCode() == 404) {
-                LOGGER.log(Level.INFO, "OpenShift OAuth got an unexpected 404 trying out the issuer's token endpoint", hre);
+            if (hre.getStatusCode() == HTTP_NOT_FOUND) {
+                LOGGER.log(INFO, "OpenShift OAuth got an unexpected 404 trying out the issuer's token endpoint", hre);
             }
-        } catch (javax.net.ssl.SSLHandshakeException ssle) {
-            LOGGER.info("OpenShift OAuth got a SSL error when accessing the issuer's token endpoint when using the SA certificate");
+        } catch (SSLHandshakeException ssle) {
+            String formattedMessage = "OpenShift OAuth got an SSL error when accessing the issuer's token endpoint when using the SA certificate";
+            LOGGER.info(formattedMessage);
             try {
+                if (jvmDefaultKeystoreTransport == null) {
```

Comment:
cross referencing with the bug's stack trace, there is no NPE at ` HttpRequestFactory requestFactory = jvmDefaultKeystoreTransport
                        .createRequestFactory(new HttpRequestInitializer() {
                        .createRequestFactory(new HttpRequestInitializer() ` a couple of lines below, which we would have gotten with our theory that `jvmDefaultKeystoreTransport` 

But I think that is expected, because of the try/catch/throwable

I don't see a full set of logs in the bugzilla @akram , but if we are correct, we should have seen 

`OpenShift OAuth provider token endpoint failed unexpectedly using the JVMs default keystore` in the log as well as the stack trace shown in the bug, which I think we would have gotten from using the default transport.

Any chance you've
a) seen the entire log and seen the log I mentioned above, or 
b) already validated this patch with the user who saw this when doing the upgrade

I'm guessing the answer is `yes` to at least one of those questions, but let me know.

Pending that clarification, I think we are good here, and we can mark this good to merge.

Reply:
Hi @gabemontero 

yes, Indeed, I saw the entire stack trace and it was showing the NPE on this value.
And, also, yes, a customer has already tested the fix for about a week.